### PR TITLE
Rebuild for CUDA 11.8 & 12.0

### PR DESCRIPTION
**Pull request by RaulPPelaez**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

This is just #57 but with the missing archs added
Closes #62 
Closes #59

### CODE REVIEW

**Comment by RaulPPelaez** [comment]
This should also make CUDA 12 work when the bot opens it.

**Comment by jakirkham** [comment]
Thanks Raul! 🙏

Would it make sense to add [the CUDA 12 migrator]( https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/acc133e6c31bc91b013f11d67414c79c421cb8eb/recipe/migrations/cuda120.yaml ) here too?

**Comment by RaulPPelaez** [comment]
Yes, did not know that was a possibility hehe

**Comment by jakirkham** [comment]
@conda-forge-admin , please re-render

**Comment by jakirkham** [comment]
> Yes, did not know that was a possibility hehe

Yep the logic is set up so that feedstocks can handle multiple migrators at once 🙂

**Comment by jakirkham** [comment]
Thanks Raul! 🙏

Looks good 👍

One last thing we could do is drop these old `skip`s. As CUDA 11.2 is the minimum in conda-forge ( https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/1708 ), these `skip`s are now unneeded

https://github.com/conda-forge/pytorch_sparse-feedstock/blob/eab47776e0f7459624e4abd70f5b3190365052a7/recipe/meta.yaml#L16-L18

**Comment by RaulPPelaez** [comment]
The CUDA 12 builds are running into a bad pybind+gcc12+CUDA12 interaction. I tried to instruct the migrator to use gcc 11 instead:
https://github.com/conda-forge/pytorch_sparse-feedstock/pull/58/commits/95f38a176303c25553b3e2469fb81a86c53801db

But its not listening to me -.-

**Comment by jakirkham** [comment]
@conda-forge-admin , please re-render

**Comment by jakirkham** [comment]
There was a new release ( https://github.com/conda-forge/pytorch_sparse-feedstock/pull/60 ), which might play better with CUDA 11.8

Tried to ask the bot to re-render, but it looks like [maintainer edits would need to be enabled]( https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests )

Alternatively could re-render manually

**Comment by RaulPPelaez** [comment]
@conda-forge-admin , please re-render

**Comment by RaulPPelaez** [comment]
Weird! the "Allow edits" box wa already checked.
The CUDA 12 migrator did not have the use_local: true flag, which was the root of the issue. I updated and rerendered, should be fine now.

**Comment by jakirkham** [comment]
> The CUDA 12 migrator did not have the use_local: true flag, which was the root of the issue. I updated and rerendered, should be fine now.

Huh weird. Thought I posted about that here somewhere, but don't see it. Sorry about that 😞

**Comment by RaulPPelaez** [comment]
Everything build, ready to merge on my side. This also includes #62 and #59, so three birds in one stone.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -26,6 +29,10 @@ requirements:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
     - {{ compiler('cuda') }}  # [cuda_compiler_version not in (undefined, 'None')]
+    {% if cuda_major >= 12 %}
+    - cuda-driver-dev
+    - cuda-nvrtc-dev
+    {% endif %}
```
Comment (jakirkham) [inline recipe/meta.yaml:35]: Think we can drop these from `build`. Having them in `host` should be sufficient

```suggestion
```

**Review by jakirkham — COMMENTED** [review]
Thanks Raul! 🙏

Had one suggestion below

**Comment by jakirkham** [comment]
Looks good. Thanks Raul 🙏

@mikemhenry when you have a chance, could you please review? 🙂

**Comment by jakirkham** [comment]
@conda-forge-admin , please re-render

**Comment by jakirkham** [comment]
Want to make sure we have fresher results (even though it passed some weeks ago)

@conda-forge-admin , please restart CI

**Comment by jakirkham** [comment]
Friendly nudge @mikemhenry 😉

**Comment by jakirkham** [comment]
@h-vetinari would you have a moment to look at this one?

File: recipe/build.sh [hunk 2]
```diff
@@ -18,6 +18,10 @@ if [[ ${cuda_compiler_version} != "None" && "$target_platform" == linux-64 ]]; t
         export TORCH_CUDA_ARCH_LIST="$TORCH_CUDA_ARCH_LIST;6.0;6.1;7.0;7.5;8.0;8.6+PTX"
```
Comment (h-vetinari) [inline recipe/build.sh:18]: Generally I'd prefer to remove settings for obsolete CUDA versions from `build.sh` wherever we migrate CUDA. But that can be done in a follow-up.

**Review by h-vetinari — APPROVED** [review]
Thanks for this @RaulPPelaez!

Please consider adding yourself as a maintainer here :)

**Pull request merged**